Mathlib/Analysis/Calculus/FDeriv/Prod.lean

```diff
@@ -487,6 +487,217 @@ theorem fderiv_pi (h : ∀ i, DifferentiableAt 𝕜 (φ i) x) :
 
 end Pi
 
+/-!
+### Derivatives of tuples `f : E → Π i : Fin n.succ, F' i`
+
+These can be used to prove results about functions of the form `fun x ↦ ![f x, g x, h x]`,
+as `Matrix.vecCons` is defeq to `Fin.cons`.
+-/
+section PiFin
+
+section TVS
+
+variable {n : ℕ} {𝕜 : Type*} {α : Fin n.succ → Type*}
+variable [Semiring 𝕜]
+variable [∀ i, AddCommMonoid (α i)] [∀ i, Module 𝕜 (α i)] [∀ i, TopologicalSpace (α i)]
+
+-- TODO: find a better home for this definition
```

Comment:
Please, also add a `LinearEquiv` version.

Reply:
Done and rehomed